java/src/main/java/com/powsybl/dataframe/network/extensions/CgmesMetadataModelDataframeAdder.java

```diff
@@ -0,0 +1,81 @@
+package com.powsybl.dataframe.network.extensions;
+
+import com.powsybl.cgmes.extensions.CgmesMetadataModelsAdder;
+import com.powsybl.cgmes.model.CgmesSubset;
+import com.powsybl.dataframe.SeriesMetadata;
+import com.powsybl.dataframe.network.adders.AbstractSimpleAdder;
+import com.powsybl.dataframe.network.adders.SeriesUtils;
+import com.powsybl.dataframe.update.IntSeries;
+import com.powsybl.dataframe.update.StringSeries;
+import com.powsybl.dataframe.update.UpdatingDataframe;
+import com.powsybl.iidm.network.*;
+
+import java.util.*;
+
+/**
+ * @author Naledi El Cheikh <naledi.elcheikh@rte-france.com>
+ */
+
+public class CgmesMetadataModelDataframeAdder extends AbstractSimpleAdder {
+    private static final List<SeriesMetadata> METADATA = List.of(
+            SeriesMetadata.stringIndex("id"),
+            SeriesMetadata.strings("subset"),
+            SeriesMetadata.strings("description"),
+            SeriesMetadata.ints("version"),
+            SeriesMetadata.strings("modelingAuthoritySet"),
+            SeriesMetadata.strings("profiles"),
+            SeriesMetadata.strings("dependentOn"),
+            SeriesMetadata.strings("supersedes")
+    );
+
+    @Override
+    public List<List<SeriesMetadata>> getMetadata() {
+        return Collections.singletonList(METADATA);
+    }
+
+    private static final class CgmesMetadataSeries {
+        private final StringSeries id;
+        private final StringSeries subset;
+        private final StringSeries description;
+        private final IntSeries version;
+        private final StringSeries modelingAuthoritySet;
+        private final StringSeries profiles;
+        private final StringSeries dependentOn;
+        private final StringSeries supersedes;
+
+        private CgmesMetadataSeries(UpdatingDataframe dataframe) {
+            this.id = dataframe.getStrings("id");
+            this.subset = dataframe.getStrings("subset");
+            this.description = dataframe.getStrings("description");
+            this.version = dataframe.getInts("version");
+            this.modelingAuthoritySet = dataframe.getStrings("modelingAuthoritySet");
+            this.profiles = dataframe.getStrings("profiles");
+            dependentOn = dataframe.getStrings("dependentOn");
+            this.supersedes = dataframe.getStrings("supersedes");
+        }
+
+        void create(int row, CgmesMetadataModelsAdder adder) {
+            CgmesMetadataModelsAdder.ModelAdder modelAdder = adder.newModel();
+            SeriesUtils.applyIfPresent(id, row, modelAdder::setId);
+            SeriesUtils.applyIfPresent(subset, row, subset -> modelAdder.setSubset(CgmesSubset.valueOf(subset)));
```

Comment:
Either we decide to stay with the long name for the python dataframe (eg. `EQUIPMENT`, `STEADY_STATE_HYPOTHESIS`) or we want the condensed form (`EQ`, `SSH`) and a new method must be added (here or in CgmesSubset) to get the right CgmesSubset from the condensed String.